components/payments/internal/connectors/plugins/public/stripe/accounts.go

```diff
@@ -0,0 +1,91 @@
+package stripe
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/formancehq/payments/internal/connectors/plugins/currency"
+	"github.com/formancehq/payments/internal/models"
+)
+
+var (
+	rootAccountReference = "root"
+)
+
+// root account reference is internal so we don't pass it to Stripe API clients
+func resolveAccount(ref *string) *string {
+	if *ref == rootAccountReference {
+		return nil
+	}
+	return ref
+}
+
+type AccountsState struct {
+	InitFinished bool   `json:"init_finished"`
+	LastID       string `json:"lastID,omitempty"`
+}
+
+func (p *Plugin) fetchNextAccounts(ctx context.Context, req models.FetchNextAccountsRequest) (models.FetchNextAccountsResponse, error) {
+	var oldState AccountsState
+	if req.State != nil {
+		if err := json.Unmarshal(req.State, &oldState); err != nil {
+			return models.FetchNextAccountsResponse{}, err
+		}
+	}
+
+	accounts := make([]models.PSPAccount, 0, req.PageSize)
+	if !oldState.InitFinished {
+		// create a root account if this is the first time this is being run
+		accounts = append(accounts, models.PSPAccount{
+			Name:      &rootAccountReference,
+			Reference: rootAccountReference,
+			CreatedAt: time.Now().UTC(),
+			Raw:       json.RawMessage("{}"),
+			Metadata:  map[string]string{},
+		})
+		oldState.InitFinished = true
+	}
+
+	needed := req.PageSize - len(accounts)
+
+	newState := AccountsState{
+		InitFinished: oldState.InitFinished,
```

Comment:
Why not setting the LastID with the oldstate LastID ?

If the call to return accounts returns none (for example last page, but empty), you will reset the state and refetch everything since the LastID on the next call will be empty

Reply:
Oops. Good catch.

Reply:
This is addressed